presto-main/src/test/java/com/facebook/presto/execution/TestCreateTableTask.java

```diff
@@ -0,0 +1,181 @@
+            throws Exception
+    {
+        CreateTable statement = new CreateTable(QualifiedName.of("test_table"),
+                ImmutableList.of(new ColumnDefinition("a", "BIGINT", Optional.empty())),
+                true,
+                ImmutableMap.of(),
+                Optional.empty());
+
+        try {
+            getFutureValue(new CreateTableTask().internalExecute(statement, metadata, new AllowAllAccessControl(), testSession, emptyList()));
+        }
+        catch (RuntimeException e) {
+            fail("CreateTableTask shouldn't throw when notExists is set to true");
+        }
+        assertEquals(metadata.getCreateTableCallCount(), 1);
+    }
+
+    @Test
+    public void testCreateTableNotExistsFalse()
+            throws Exception
+    {
+        CreateTable statement = new CreateTable(QualifiedName.of("test_table"),
+                ImmutableList.of(new ColumnDefinition("a", "BIGINT", Optional.empty())),
+                false,
+                ImmutableMap.of(),
+                Optional.empty());
+
+        try {
+            getFutureValue(new CreateTableTask().internalExecute(statement, metadata, new AllowAllAccessControl(), testSession, emptyList()));
+            fail("CreateTableTask should throw when notExists is set to false");
+        }
+        catch (RuntimeException e) {
+            // Expected
+        }
+        assertEquals(metadata.getCreateTableCallCount(), 1);
+    }
+
+    private static class MockMetadata
+            extends AbstractMockMetadata
+    {
+        private final TypeManager typeManager;
+        private final TablePropertyManager tablePropertyManager;
+        private final ConnectorId catalogHandle;
+        private int createTableCallCount;
+
+        public MockMetadata(
+                TypeManager typeManager,
+                TablePropertyManager tablePropertyManager,
+                ConnectorId catalogHandle)
+        {
+            this.typeManager = requireNonNull(typeManager, "typeManager is null");
+            this.tablePropertyManager = requireNonNull(tablePropertyManager, "tablePropertyManager is null");
+            this.catalogHandle = requireNonNull(catalogHandle, "catalogHandle is null");
+        }
+
+        @Override
+        public void createTable(Session session, String catalogName, ConnectorTableMetadata tableMetadata)
+        {
+            createTableCallCount++;
+            throw new PrestoException(ALREADY_EXISTS, "Table already exists");
+        }
+
+        @Override
+        public TablePropertyManager getTablePropertyManager()
+        {
+            return tablePropertyManager;
+        }
+
+        @Override
+        public Type getType(TypeSignature signature)
+        {
+            return typeManager.getType(signature);
+        }
+
+        @Override
+        public Optional<ConnectorId> getCatalogHandle(Session session, String catalogName)
+        {
+            return Optional.of(catalogHandle);
```

Comment:
Let's check if the catalog name matches the name in the handle and return empty if not.